### PR TITLE
refactor(metadata): replace HttpOperation constants by raw strings

### PR DESCRIPTION
**Pull request by MarvinCourcier**
As spoken with @chalasr there are string hardcoded operation request which could be replaced with the HttpOperation request const

Maybe this isn't what you want but I thought this would be good to replace those.
I'll gladly take any advice.

I don't know to which branch this should be merged so I'd like your advice.

### CODE REVIEW

**Comment by soyuka** [comment]
Indeed this is a good change, you should change your commit though:

```
refactor(metadata): use HttpOperation constants for methods
```

`chore` is used for cleanup or things you can't categorize.

**Comment by MarvinCourcier** [comment]
> Indeed this is a good change, you should change your commit though:
> 
> ```
> refactor(metadata): use HttpOperation constants for methods
> ```
> 
> `chore` is used for cleanup or things you can't categorize.

Sur thing ! it's done !

**Comment by soyuka** [comment]
@MarvinCourcier the commit should be changed, not the Pull Request title which is unrelated after merging except for github.

**Comment by dunglas** [comment]
Could you do a profile please? IIRC resolving the constants has a runtime cost that can be significant and it's why we do not use them at least on the hot path.

**Comment by dunglas** [comment]
👍 for consistency, but let's use a raw string everywhere then.

File: src/Metadata/HttpOperation.php [hunk 1]
```diff
@@ -18,14 +18,6 @@
 
 class HttpOperation extends Operation
 {
-    public const METHOD_GET = 'GET';
-    public const METHOD_POST = 'POST';
-    public const METHOD_PUT = 'PUT';
-    public const METHOD_PATCH = 'PATCH';
-    public const METHOD_DELETE = 'DELETE';
-    public const METHOD_HEAD = 'HEAD';
-    public const METHOD_OPTIONS = 'OPTIONS';
```
Comment (MarvinCourcier) [inline src/Metadata/HttpOperation.php:27]: Since we want to use raw string, I don't see the point of those constant to remain.
So I've deleted them

Comment (MarvinCourcier) [reply src/Metadata/HttpOperation.php:27]: Let's ask to @dunglas and @chalasr then, what should be done there ?

How do you wish to proceed ?

Comment (chalasr) [reply src/Metadata/HttpOperation.php:27]: Reasoning is: these constants are for end users, api platform does not reference them for performance reasons. They must stay, and the codebase should not use them at all for consistency as per @dunglas statement.

Comment (dunglas) [reply src/Metadata/HttpOperation.php:27]: I totally agree with @chalasr

Comment (MarvinCourcier) [reply src/Metadata/HttpOperation.php]: It's done then, I've reverted it back the way it was :)

**Comment by soyuka** [comment]
thanks

**Pull request merged**